packages/libp2p/test/circuit-relay/discovery.node.ts

```diff
@@ -34,6 +35,7 @@ describe('circuit-relay discovery', () => {
         mockContentRouting()
       ],
       services: {
+        identify: identifyService(),
```

Comment:
Why was it necessary to add identify as part of config validation?  I see this change in a lot of tests here.

Reply:
Ah, I see - it's because you've added validation to the `validateConfig` method to enforce the presence of certain service keys.

Identify is only necessary for circuit relay for ambient relay discovery (e.g. after dialing a peer we notice they are a relay and try to get a reservation if we need one).

In this test we ensure that the content routing module can find relays (e.g. via searching for the advertisement CID) so we actually don't want identify enabled.

Anyway I think enforcing service dependencies is outside of the scope of this PR, please revert in https://github.com/libp2p/js-libp2p/pull/2133.

Reply:
We can discuss the use cases [here](https://github.com/libp2p/js-libp2p/discussions/2135)